File: lib/models/destiny_item_perks_component.dart
import 'destiny_perk_reference.dart';
class DestinyItemPerksComponent{
	List<DestinyPerkReference> perks;
	DestinyItemPerksComponent(
		this.perks,
	);

	static DestinyItemPerksComponent fromMap(Map<String, dynamic> data){
		if(data == null) {
			return null;
		};
		return new DestinyItemPerksComponent(
				data['perks'] != null ? DestinyPerkReference.fromList(data['perks']) : null,
		);
	}

	static List<DestinyItemPerksComponent> fromList(List<dynamic> data){
		if(data == null) {
			return null;
		};
		List<DestinyItemPerksComponent> list = new List();
    data.forEach((item) {
      list.add(DestinyItemPerksComponent.fromMap(item));
    });
    return list;
	}

	Map<String, dynamic> toMap(){
		Map<String, dynamic> data = new Map();
			data['perks'] = this.perks.map((item)=>item.toMap()).toList();
		return data;
	}
}

File: lib/src/models/DestinyItemPerksComponent.cs
using System.Runtime.Serialization;
using System.Collections.Generic;

namespace BungieNetApi.Model {
	/// Instanced items can have perks: benefits that the item bestows.
	/// These are related to DestinySandboxPerkDefinition, and sometimes - but not always - have human readable info. When they do, they are the icons and text that you see in an item&#39;s tooltip.
	/// Talent Grids, Sockets, and the item itself can apply Perks, which are then summarized here for your convenience.
	[DataContract]
	public class DestinyItemPerksComponent{

		/// <summary>
		/// The list of perks to display in an item tooltip - and whether or not they have been activated.
		/// </summary>
		[DataMember(Name="perks", EmitDefaultValue=false)]
		public List<DestinyPerkReference> Perks { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyItemPerksComponent);
        }

		public bool Equals(DestinyItemPerksComponent input)
		{
			if (input == null) return false;

			return
				(
                    Perks == input.Perks ||
                    (Perks != null && Perks.Equals(input.Perks))
                ) ;
		}
	}
}

